Fix makepdf (script which creates the pdf)

mmattel opened the pull request:
`makepdf` is improved similar as we have it now in the `docs-client-desktop` repo.
This eases the setup for the new docs structure a lot as we just need to copy/paste things now and set the right variables in the script!

An additional fix has been made in site.yml pointing to the correct docs domain...
@tbsbdr @jnweiger fyi

Backport to 10.7 and 10.6 necessary

Info, in the desktop repo, we already have a instruction manual for this. Will improve it first and copy it afterwards.

phil-davis commented on bin/makepdf:
```diff
@@ -1,5 +1,14 @@
+# Each manual build is based on a book template found in books. It is mandatory to be present
+# This script creates pdf versions of the manual
+# If no manual is in the parameter list, all available manuals are built to pdf
+# If called from the command line, the script user the term "master" as reference
```
```suggestion
# If called from the command line, the script uses the term "master" as reference
```

And we might want to stop using the term "master" where possible.

EParzefall replied:
For political correctness reasons I assume? Should be easy here, but that raises the question if we should rename our master branches to 'main'.

phil-davis replied:
I raised issue https://github.com/owncloud/enterprise/issues/4560 for discussing about that.
I just thought that we may as well not add more uses of the word "master".
Example of the industry discussion: https://thenewstack.io/words-matter-finally-tech-looks-at-removing-exclusionary-language/

phil-davis commented on bin/makepdf:
```diff
@@ -1,5 +1,14 @@
+# If called from the command line, the script user the term "master" as reference
+# If called from drone, the script uses additional variables (like a branch name)
+# The script uses another script to provide the attribute list of the yaml file dynamically
+# Aware, that a manual name MUSt NOT contain a blank
```
```suggestion
# Be aware that a manual name MUST NOT contain a blank
```

phil-davis commented on bin/makepdf:
```diff
@@ -146,23 +88,26 @@ function convert_antora_nav_to_asciidoc_list()
-        echo "[${manual}] is not a valid manual." 
-        echo "Available manuals are admin, developer, and user."
-        return $ERR_UNSUPPORTED_MANUAL
+    # ok if manual is is list of possible manuals
```
```suggestion
    # ok if manual is in the list of possible manuals
```

phil-davis commented on bin/makepdf:
```diff
@@ -184,61 +129,53 @@ function build_pdf_manual()
 {
+    local actual_manual
+
+    # if no specific manual is named, built them all
```
```suggestion
    # if no specific manual is named, then build them all
```

phil-davis reviewed (commented):
Just a few words.

EParzefall reviewed (approved):
Approved once you incorporated Phil's suggestions and replaced 'master' with something more suitable.

mmattel commented:
Thanks for the suggestions, will incorporate them also in the desktop repo script.
I will update the script to make `master` a variable so it can be exchanged later easily

phil-davis reviewed (approved):
Looks good now. The `master` thing is for another issue and another day (month, year?)

mmattel commented:
* Added a description file `bin/instructions_makepdf.md` for `makepdf` (any text changes will also be used in the desktop repo). This is currently not linked and I think that is not necessary to do so.
* Improved the script to use the "master" branch name as variable which eases later branch renames to eg "main"
* Added a comment line in one document because use of the kroki extension (see added text for more details)

mmattel commented on bin/instructions_makepdf.md:
```diff
@@ -0,0 +1,98 @@
+
+## How a pdf is Built
+
+The general idea of `asciidoctor-pdf` is to create a pdf from a given single file and add options to the build command, see the [CLI Options](https://docs.asciidoctor.org/asciidoctor.js/latest/cli/options/) for details. Because the manuals are usually made out of multiple documents, a base document must be created which is then built to pdf. Therefore following method is used: A book template fil ...
```
```suggestion
The general idea of `asciidoctor-pdf` is to create a pdf from a given single file and add options to the build command, see the [CLI Options](https://docs.asciidoctor.org/asciidoctor.js/latest/cli/options/) for details. Because manuals are usually made out of multiple documents, a base document must be created which is then built to pdf. Therefore following method is used: A book template file has to be physically present in the `books/` folder. The following example file shows the basic structure and provides some settings.
```

EParzefall commented on bin/instructions_makepdf.md:
```diff
@@ -0,0 +1,98 @@
+
+## Build Errors
+
+In very rare cases, it can happen that a pdf build errors. This should not be neglected and resolved. One issue identified (and solved) was a case when using a multi line source bash code block which trailed with a `\`. This  code block would not work in bash, should render properly but did not due to a backend bug, see [pdf creation fails when using a codeblock that ends with \ but has no follow ...
```
```suggestion
In very rare cases, it can happen that a pdf build produces errors. This should not be neglected but resolved. One issue identified (and solved) was a case when using a multi-line source bash code block which trailed with a `\`. This  code block would not work in bash, should render properly but did not due to a backend bug, see [pdf creation fails when using a codeblock that ends with \ but has no following line with content](https://github.com/asciidoctor/asciidoctor-pdf/issues/1930). In such cases, use the debug mode `-d`, take the file and create a manual build on the command line, adding all attributes and commenting out all includes. Step by step re-enable them and restart the build to see which document is failing. When the problematic file is found, make a copy of the file for backup purposes. Delete all content up to the first section and restart building. Copy back section by section and always do a build. When the problematic section is found do the same thing with the section content. When the issue has been identified, fix it to see if it will build successfully. If a pattern is identifiable, fix all instances and provide those fixes in the backup file. Finally you can drop the debug file and rename the backup file to its original name. A normal build should now succeed.
```

EParzefall replied:
Oops, sorry the changes aren't showing up. Maybe too large a block?

EParzefall commented on bin/makepdf:
```diff
@@ -1,5 +1,14 @@
 #!/usr/bin/env bash
 
+# Use and adopt this script only if there are multiple manuals
```
```suggestion
# Use and adapt this script only if there are multiple manuals
```

EParzefall commented on bin/makepdf:
```diff
@@ -1,5 +1,14 @@
 #!/usr/bin/env bash
 
+# Use and adopt this script only if there are multiple manuals
+# Each manual build is based on a book template found in the directory books. It is mandatory to be present
```
```suggestion
# Each manual build is based on a book template found in the mandatory directory books.
```

EParzefall commented on bin/makepdf:
```diff
@@ -1,5 +1,14 @@
 
+# Use and adopt this script only if there are multiple manuals
+# Each manual build is based on a book template found in the directory books. It is mandatory to be present
+# This script creates pdf versions of the manual
+# If no manual is in the parameter list, all available manuals are built to pdf
+# If called from the command line, the script user the term "master" (or "main") as branch reference
+# If called from drone, the script uses additional variables (like a branch name)
+# The script uses another script to provide the attribute list of the yaml file dynamically
+# Be aware that a manual name MUST NOT contain a blank
```
```suggestion
# This script creates pdf versions of the manual.
# If no manual is in the parameter list, all available manuals are built to pdf.
# If called from the command line, the script user the term "master" (or "main") as branch reference.
# If called from drone, the script uses additional variables (like a branch name).
# The script uses another script to provide the attribute list of the yaml file dynamically.
# Be aware that a manual name MUST NOT contain a blank.
```

EParzefall commented on bin/makepdf:
```diff
@@ -146,30 +90,35 @@ function convert_antora_nav_to_asciidoc_list()
     # Get the dynamic list of attributes from site.yml
     # The output after sed is a string like -a key=value -a key=value ...
     # For testing, also use the yamltest.sh script
+    # You can do the same for extension like kroki if needed (currently not implemented)
```
```suggestion
    # You can do the same for extensions like kroki if needed (currently not implemented).
```

EParzefall commented on bin/makepdf:
```diff
@@ -184,61 +133,53 @@ function build_pdf_manual()
+    # https://docs.asciidoctor.org/asciidoctor.js/latest/cli/options/
+
     # Create argument list, necessary as we have dynamic attributes coming from another file
     # The param string needs to be proper constructed, pls be careful, see comment below
```
```suggestion
    # The param string needs to be properly constructed, please be careful, see comment below.
```

EParzefall commented on modules/admin_manual/pages/configuration/server/security/oauth2.adoc:
```diff
@@ -18,6 +18,7 @@ The OAuth 2.0 authorization framework enables a third-party application to obtai
 
 Here is an overview of how the process works:
 
+// Based on the kroki extension. Marked for ease of finding. Be aware that this renders by default to svg which has issues when building pdf. ATM, this is not an issue as this document is not part of the pdf (too deep in the level limit) and because the extension is not added to the pdf build call. This would need an adoption of the script. Be aware that tabs.js MUST be excluded in case (html onl ...
```
```suggestion
// Based on the kroki extension. Marked for ease of finding. Be aware that this renders by default to svg which has issues when building pdf. ATM, this is not an issue as this document is not part of the pdf (too deep in the level limit) and because the extension is not added to the pdf build call. This would need an adjustment of the script. Be aware that tabs.js MUST be excluded in case (html only)! https://github.com/Mogztter/asciidoctor-kroki
```

EParzefall replied:
Last sentence before the link: in case of what? Maybe: in case you want to try?

EParzefall reviewed (approved):
Some language fixes and some questions remain to be dealt with.

mmattel commented on bin/instructions_makepdf.md:
```diff
@@ -0,0 +1,98 @@
+-c ... clean the build/ directory (contains the pdf)
+-d ... debug mode, print out with what it would work, only with -m
+-m ... build all modules
+-n ... build module <name>, only with -m
```
only in combination with -m

The script has been build by the originators to provide the help text when no parameters are given. You are right, this needs better text

mmattel replied:
fixed

mmattel commented on bin/instructions_makepdf.md:
```diff
@@ -0,0 +1,98 @@
+
+-h ... help
+-c ... clean the build/ directory (contains the pdf)
+-d ... debug mode, print out with what it would work, only with -m
+-m ... build all modules
+-n ... build module <name>, only with -m
```
```suggestion
-d ... Debug mode, prints the book to be converted. Only in combination with -m and/or -n
-m ... Build all available manuals
-n ... Build manual <name>. Only in combination with -m
```

mmattel commented on bin/makepdf:
```diff
@@ -37,13 +46,24 @@ if [[ -z "${VERSION:-}" ]]; then
+    echo "Usage: ./bin/makepdf [-c] [-d] [-h] [-m] [-n <${list}>]"
+    echo
+    echo "-h ... help"
+    echo "-c ... clean the build/ directory (contains the pdf)"
+    echo "-d ... debug mode, print out with what it would work, only with -m"
+    echo "-m ... build all modules"
+    echo "-n ... build module <name>, only with -m"
```
```suggestion
    echo "-c ... clean the build/ directory (contains the pdf)"
    echo "-d ... Debug mode, prints the book to be converted. Only in combination with -m and/or -n"
    echo "-m ... Build all available manuals"
    echo "-n ... Build manual <name>. Only in combination with -m"
```
